### PR TITLE
Convert home-assistant-main to ES6 class

### PR DESCRIPTION
**Pull request by emlove**

### CODE REVIEW

File: src/layouts/home-assistant-main.html [hunk 1]
```diff
@@ -87,83 +88,89 @@
...
+        dockedSidebar: {
+          type: Boolean,
+          computed: 'computeDockedSidebar(hass)',
+        },
+      };
+    }
+
+    ready() {
+      super.ready();
+      this.addEventListener('hass-open-menu', this.handleOpenMenu);
```
Comment (andrey-git) [inline src/layouts/home-assistant-main.html]: Is `this` auto-bound to those callbacks?

Comment (armills) [reply src/layouts/home-assistant-main.html]: Good question. They seem to be working in the browser, although I'm trying to find some documentation that supports this is how ES6 classes work.

Comment (armills) [reply src/layouts/home-assistant-main.html]: I'm just going to add arrow functions to be safe.

Comment (andrey-git) [reply src/layouts/home-assistant-main.html]: Here: https://stackoverflow.com/questions/36489579/this-within-es6-class-method
It says that `this` of event listeners is the object that got the event, which in this case is the same `home-assistant`

File: src/layouts/home-assistant-main.html [hunk 2]
```diff
@@ -87,83 +88,89 @@
...
+          computed: 'computeDockedSidebar(hass)',
+        },
+      };
+    }
+
+    ready() {
+      super.ready();
+      this.addEventListener('hass-open-menu', () => this.handleOpenMenu());
+      this.addEventListener('hass-close-menu', () => this.handleCloseMenu());
+      this.addEventListener('hass-start-voice', () => this.handleStartVoice());
```
Comment (andrey-git) [inline src/layouts/home-assistant-main.html]: pass in `ev` that handleStartVoice needs

**Pull request merged**